mesonbuild/dependencies/base.py

```diff
@@ -916,12 +916,17 @@ class CMakeDependency(ExternalDependency):
     # multiple times in the same Meson invocation.
     class_cmakebin = PerMachine(None, None, None)
     class_cmakevers = PerMachine(None, None, None)
+    class_cmakeinfo = PerMachine(None, None, None)
     # We cache all pkg-config subprocess invocations to avoid redundant calls
     cmake_cache = {}
     # Version string for the minimum CMake version
     class_cmake_version = '>=3.4'
     # CMake generators to try (empty for no generator)
     class_cmake_generators = ['', 'Ninja', 'Unix Makefiles', 'Visual Studio 10 2010']
+    class_working_generator = None
+    # Cache for os.listdir and os.path.exists
+    class_listdir_cache = {}
+    class_isdir_cache = {}
```

Comment:
Instead of creating a cache for each of these, you make the functions that use these class methods and use `@functools.lru_cache()`. It's simpler, as long as you make sure that you consume and return immutable types.

Reply:
Sure, thanks for the tip. Updated.